src/Umbraco.Web.Common/Extensions/HttpRequestExtensions.cs

```diff
@@ -52,6 +55,21 @@ private static bool IsSet(this IPAddress address)
         {
             const string NullIpAddress = "::1";
             return address != null && address.ToString() != NullIpAddress;
+        }
+
+        public static string GetRawBodyString(this HttpRequest request, Encoding encoding = null)
```

Comment:
Should there be an async version of this?  Then the JsonPathBinder could use the async version in its BindModelAsync method.

Reply:
Added one